pkgs/development/python-modules/pomegranate/default.nix

```diff
@@ -5,34 +5,54 @@
 , fetchpatch
 , numpy
 , scipy
-, cython
+, torch
 , networkx
-, joblib
-, pandas
+, apricot-select
+, scikit-learn
+, pytestCheckHook
 , nose
-, pyyaml
 }:
 
 
 buildPythonPackage rec {
   pname = "pomegranate";
-  version = "0.14.8";
+  version = "1.0.0";
 
   src = fetchFromGitHub {
     repo = pname;
     owner = "jmschrei";
-    # no tags for recent versions: https://github.com/jmschrei/pomegranate/issues/974
-    rev = "0652e955c400bc56df5661db3298a06854c7cce8";
-    sha256 = "16g49nl2bgnh6nh7bd21s393zbksdvgp9l13ww2diwhplj6hlly3";
+    rev = "v${version}";
+    hash = "sha256-EnxKlRRfsOIDLAhYOq7bUSbI/NvPoSyYCZ9D5VCXFGQ=";
   };
 
-  propagatedBuildInputs = [ numpy scipy cython networkx joblib pyyaml ];
+  propagatedBuildInputs = [ numpy scipy torch networkx scikit-learn apricot-select ];
 
-  nativeCheckInputs = [ pandas nose ];  # as of 0.13.5, it depends explicitly on nose, rather than pytest.
+  nativeCheckInputs = [ nose pytestCheckHook ];
+
+  # These tests seem to be broken as of 1.0.0
```

Comment:
The full PR looks good to me.

Only interested in what errors the disabled tests are giving. Anything worth reporting upstream?

Reply:
Upstream disables `sample` in their own workflows, the other tests should be reported upstream, imo. These are the errors I get:

<details>

<summary>Summary of errors</summary>

```sh
FAILED tests/test_bayesian_network.py::test_learn_structure_chow_liu - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network.py::test_learn_structure_exact - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_chow_liu - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_chow_liu_weighted - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_chow_liu_large - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_chow_liu_large_pseudocount - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_exact - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_exact_weighted - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_exact_exclude_parents - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_exact_large - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_exact_large_pseudocount - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_learn_structure_chow_liu - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
FAILED tests/test_bayesian_network_structure_learning.py::test_categorical_learn_structure_exact - ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape...
```

</details>

Reply:
Upstream have fixed these tests in master, but they haven't yet released a version with the fixed tests. The good news is that the only changes made in master are to the tests themselves. I will bump the version when they issue a release.